plugins/CoreConsole/Commands/DevelopmentSyncUITestScreenshots.php

```diff
@@ -73,42 +71,166 @@ protected function execute(InputInterface $input, OutputInterface $output)
             if ($file !== null
                 && preg_match("/" . $screenshotsRegex . "/", $file)
             ) {
-                if ($testPlugin == null) {
-                    $downloadTo = "tests/UI/expected-ui-screenshots/$file";
-                } else {
-                    $downloadTo = "plugins/$testPlugin/tests/UI/expected-ui-screenshots/$file";
-                }
-
-                $output->write("<info>Downloading $file to  $downloadTo...</info>\n");
-                Http::sendHttpRequest("$urlBase/processed-ui-screenshots/$file", $timeout = 60, $userAgent = null,
-                    PIWIK_DOCUMENT_ROOT . "/" . $downloadTo);
+                $this->downloadProcessedScreenshot($input, $output, $file);
             }
         }
 
-        $this->displayGitInstructions($output);
+        $this->displayGitInstructions($input, $output);
 
     }
 
-    /**
-     * @param OutputInterface $output
-     */
-    protected function displayGitInstructions(OutputInterface $output)
+    protected function displayGitInstructions(InputInterface $input, OutputInterface $output)
```

Comment:
Small detail: you might want to get `$plugin` and pass it to this function instead of InputInterface. This would keep the argument/option reading in one place.

(Alt text: maybe you could pass the `$plugin` into this function instead of InputInterface? then the argument/option reading would be in one place.)